tests/test_python.py

```diff
@@ -187,6 +187,17 @@ def test_tensorf(container_per_test):
     # check the expected CPU flag for TF is available in the system
     container_per_test.connection.run_expect([0], 'lscpu| grep -i " SSE4"')
 
+    # pip main version number 
+    pip_version = container_per_test.connection.run("pip --version|tr -cd .0-9| cut -d. -f1").stdout.strip(" \n")
+
+    # python tag version number
+    python_version = container_per_test.connection.run("python3 --version|tr -cd .0-9| cut -d. -f1,2").stdout.strip(" \n")
+
+    if python_version == "3.10":
+       if int(pip_version) <= 20:
+           # pip upgrade needed
+           pytest.xfail("To install Tensorflow with python 3.10 use a pip version > 20")
```

Comment:
@jlausuch From your experience as a test writer, do you think we should implement the xfail like this or just xfail if `pip install tensorflow` fails? I see advantages & issues with both so what's your take on this?

Reply:
The current implementation is a pre-analysis of the failure for a specific case, 
but indeed a simpler check of pip install return value could intercept more cases with less code, eventually needing investigation on failure.
I will apply your suggestion.

Reply:
It's a good question. Both have their own nits... However, I agree the second option could catch other errors in the future (not even related to pip) when installing Tensorflow, which is the main use case here.